AwsEncryptionSDK/runtimes/net/ESDK.csproj

```diff
@@ -29,10 +29,11 @@
   </PropertyGroup>
 
   <ItemGroup>
-      <PackageReference Include="AWSSDK.Core" Version="3.7.103" />
+      <!-- TODO: manually upgraded to match the latest from MPL, is that reasonable? -->
+      <PackageReference Include="AWSSDK.Core" Version="3.7.300.2" />
       <PackageReference Include="DafnyRuntime" Version="4.2.0" />
       <PackageReference Include="BouncyCastle.Cryptography" Version="2.2.1" />
-      <PackageReference Include="AWS.Cryptography.MaterialProviders" Version="1.0.0" />
+      <ProjectReference Include="../../../mpl/AwsCryptographicMaterialProviders/runtimes/net/MPL.csproj" />
```

Comment:
This works for local development, but I think this will break when published.
There is an `if` condition that can be added to account for this,
but I think that introduces a way for installed versions of the ESDK
to be able to be manipulated by placing a custom project in a specific place.

@josecorella We talked about this right? I think you did some testing?

Reply:
Shouldn't the submodule be in exactly the 1.0.0 published state though, so this is equivalent?

We already have the same thing in the DB ESDK: https://github.com/aws/aws-database-encryption-sdk-dynamodb/blob/main/DynamoDbEncryption/runtimes/net/DynamoDbEncryption.csproj#L38

Reply:
Ahh, yes. This is a double edged sword. It makes local development nice because you can test out local changes and in flight changes. 

@robin-aws is right that this all depends on the place of the submodule, it this points to 1.0.0 or the equivalent version this is fine. 

As long as we don't bump the submodule to something other than the public dependency, this is the preferred method of using "local" dependencies according to the .net sdk team.

This allows us to test incoming breaking changes in other branches by just updating the submodule pointer without affecting how we depend on the MPL.

Reply:
OK, looks good to me then. I would like to know what our long term mechanism is to make sure that the submodule does point to published version.